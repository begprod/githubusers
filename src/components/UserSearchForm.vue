<template>
<form @submit.prevent="$emit('submitted', username)">
  <input type="text" v-model="username">
  <button>Submit</button>
</form>
</template>

<script>
export default {
  name: 'UserSearchForm',
  data () {
    return {
      username: ''
    }
  }
}
</script>
