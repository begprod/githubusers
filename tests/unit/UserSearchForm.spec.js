import { shallowMount } from '@vue/test-utils'
import UserSearchForm from '@/components/UserSearchForm'

describe('User search form', () => {
  const build = () => {
    const wrapper = shallowMount(UserSearchForm)
    return {
      wrapper,
      input: () => wrapper.find('input'),
      button: () => wrapper.find('button')
    }
  }
  it('renders the component', () => {
    const { wrapper } = build()
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('render child components', () => {
    const { input, button } = build()
    expect(input().exists()).toBe(true)
    expect(button().exists()).toBe(true)
  })
  it('call submit event', () => {
    const expectUser = 'begprod'
    const { wrapper, input, button } = build()
    input().element.value = expectUser
    input().trigger('input')
    button().trigger('click')
    button().trigger('submit')
    expect(wrapper.emitted().submitted[0]).toEqual([expectUser])
  })
})
