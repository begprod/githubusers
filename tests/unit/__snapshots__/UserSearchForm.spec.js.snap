// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User search form renders the component 1`] = `<form><input type="text"> <button>Submit</button></form>`;
